components/Table/Table.tsx

```diff
@@ -70,161 +70,163 @@ type TableProps = {
-									{...column.getHeaderProps(
-										column.sortable ? column.getSortByToggleProps() : undefined
-									)}
-								>
-									{column.render('Header')}
-									{column.sortable && (
-										<SortIconContainer>
-											{column.isSorted ? (
-												column.isSortedDesc ? (
-													<StyledSortDownIcon />
+		return (
+			<>
+				<TableContainer>
+					<ReactTable {...getTableProps()} palette={palette} className={className}>
+						{headerGroups.map((headerGroup) => (
+							<div className="table-row" {...headerGroup.getHeaderGroupProps()}>
+								{headerGroup.headers.map((column: any) => (
+									<TableCellHead
+										hideHeaders={hideHeaders}
+										{...column.getHeaderProps(
+											column.sortable ? column.getSortByToggleProps() : undefined
+										)}
+									>
+										{column.render('Header')}
+										{column.sortable && (
+											<SortIconContainer>
+												{column.isSorted ? (
+													column.isSortedDesc ? (
+														<StyledSortDownIcon />
+													) : (
+														<StyledSortUpIcon />
+													)
 												) : (
-													<StyledSortUpIcon />
-												)
-											) : (
-												<>
-													<StyledSortUpIcon />
-													<StyledSortDownIcon />
-												</>
-											)}
-										</SortIconContainer>
-									)}
-								</TableCellHead>
-							))}
-						</div>
-					))}
-					{isLoading ? (
-						<StyledSpinner />
-					) : (
-						page.length > 0 && (
-							<TableBody className="table-body" {...getTableBodyProps()}>
-								{page.map((row: Row, idx: number) => {
-									prepareRow(row);
-									const props = row.getRowProps();
-									const localRef = lastRef && idx === page.length - 1 ? lastRef : defaultRef;
-									const handleClick = onTableRowClick ? () => onTableRowClick(row) : undefined;
-									return (
-										<TableBodyRow
-											localRef={localRef}
-											highlightRowsOnHover={highlightRowsOnHover}
-											row={row}
-											onClick={handleClick}
-											{...props}
-										/>
-									);
-								})}
-							</TableBody>
-						)
-					)}
-					{!!noResultsMessage && !isLoading && data.length === 0 && noResultsMessage}
-				</ReactTable>
-			</TableContainer>
-			{showPagination && !showShortList && data.length > (pageSize ?? MAX_PAGE_ROWS) ? (
-				<Pagination
-					compact={compactPagination}
-					pageIndex={pageIndex}
-					pageCount={pageCount}
-					canNextPage={canNextPage}
-					canPreviousPage={canPreviousPage}
-					setPage={gotoPage}
-					previousPage={previousPage}
-					nextPage={nextPage}
-				/>
-			) : undefined}
-		</>
-	);
-};
+													<>
+														<StyledSortUpIcon />
+														<StyledSortDownIcon />
+													</>
+												)}
+											</SortIconContainer>
+										)}
+									</TableCellHead>
+								))}
+							</div>
+						))}
+						{isLoading ? (
+							<StyledSpinner />
```

Comment:
Could be good to replace with our `Loader` component

Reply:
Done.